rust/datafusion/src/optimizer/hash_build_probe_order.rs

```diff
@@ -55,7 +53,25 @@ fn get_num_rows(logical_plan: &LogicalPlan) -> Option<usize> {
             let num_rows_input = get_num_rows(input);
             num_rows_input.map(|rows| std::cmp::min(*limit, rows))
         }
-        _ => None,
+        LogicalPlan::Aggregate { .. } => {
+            // we cannot yet predict how many rows will be produced by an aggregate because
+            // we do not know the cardinality of the grouping keys
+            None
+        }
+        LogicalPlan::Filter { .. } => {
+            // we cannot yet predict how many rows will be produced by a filter because
+            // we don't know how selective it is (how many rows it will filter out)
+            None
+        }
+        _ => {
```

Comment:
Looks good. I wonder if it is better to explicitly match on all arms, so if we add a new LP alternative we have to think about the implications here.

Reply:
Yes, I think that would be better.